content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md

```diff
@@ -0,0 +1,144 @@
+
+To view data object, follow this basic procedure:
+1. Establish a link to the data object.
+2. Make a command that the data object will follow.
+3. Consider the data that the data object has provided.
+
+To complete the actions listed above, we will need to employ the following classes:
+1. System.Data.SqlClient.SqlConnection
+2. System.Data.SqlClient.SqlCommand
+3. System.Data.SqlClient.SqlDataReader
+
+### Preparation
+For this project, I used a safe console application. To begin, we must declare the variables that will be used. We will need to make a variable to hold the data, one to hold the connection attributes, and another to hold the SQL statements we will be issuing. Any exceptions that we encounter when reading from the database are noted. In the working-storage section, all of these items are listed below.
+
+```SQL
+working-storage section.
+sqlConnection type System.Data.SqlClient.SqlConnection.
+sqlDataReader type System.Data.SqlClient.SqlDataReader.
+sqlCommand    type System.Data.SqlClient.SqlCommand.
+```
+These are the essential objects that will allow us to connect to the database engine and use ADO, as well as get information from the database engine about the operations we run there.
+
+A few functional fields will be added next, as well as a few fields for storing data we'll be retrieving from the database. The following are the contents of the completed working-storage section:
+
+
+```sql
+working-storage section.
+sqlConnection type System.Data.SqlClient.SqlConnection.
+sqlDataReader type System.Data.SqlClient.SqlDataReader.
+sqlCommand    type System.Data.SqlClient.SqlCommand.
+
+sqlCmd             string.
+dataFound          type System.Boolean.
+recordsRead        pic 9(06)  value zeros.
+LastName           pic x(20)  value spaces.
+FirstName          string.
+HomePhone          pic x(24)  value spaces.    
+junk               pic x(01)  value spaces.
+```
+
+It's worth noting that I can mix and match the types of data definitions. Some variables were defined using typical COBOL syntax for the PIC clause, while others were defined using standard .NET data types. By enabling this option, some of your source code can now be used without worrying about data type conversion. To read data from a table using ADO, we've generated all of the necessary divisions and references. We must now carry out the measures mentioned above.
+
+### Create a Connection
+In order to use a database, we must first establish a connection. If you've ever used ODBC with embedded SQL, you'll know that you must first establish the connection in the Control Panel's ODBC administrators panel before issuing the SQL connect query. "EXEC SQL CONNECT TO DATABASE END-EXEC" was comparable to the connect statement.

```

Comment:
Where do you find this `Control Panel's ODBC administrators panel`?

Add image